File: src/main/java/mnm/mods/tabbychat/client/gui/ChatHandle.java
package mnm.mods.tabbychat.client.gui;

import mnm.mods.tabbychat.util.Color;
import mnm.mods.tabbychat.util.Dim;
import mnm.mods.tabbychat.util.ILocation;
import mnm.mods.tabbychat.util.Location;
import mnm.mods.tabbychat.client.gui.component.GuiComponent;

import javax.annotation.Nonnull;

public class ChatHandle extends GuiComponent {

    ChatHandle() {
        setLocation(new Location(0, 0, 10, 10));
    }

    @Override
    public void render(int mouseX, int mouseY, float parTicks) {
        ILocation loc = getLocation();
        int color = getColor(loc.contains(mouseX, mouseY)).getHex();
        this.drawHorizontalLine(loc.getXPos() + 3, loc.getXWidth() - 4, loc.getYPos() + 4, color);
        this.drawVerticalLine(loc.getXWidth() - 4, loc.getYPos() + 4, loc.getYHeight() - 2, color);
    }

    @Nonnull
    private Color getColor(boolean hovered) {
        int opac = (int) (mc.gameSettings.chatOpacity * 255);
        return Color.of(255, 255, hovered ? 160 : 255, opac);
    }

    @Nonnull
    @Override
    public Dim getMinimumSize() {
        return new Dim(12, 12);
    }
}

File: src/main/kotlin/mnm/mods/tabbychat/client/gui/ChatHandle.kt
package mnm.mods.tabbychat.client.gui

import mnm.mods.tabbychat.util.Color
import mnm.mods.tabbychat.util.Colors
import mnm.mods.tabbychat.util.argb
import mnm.mods.tabbychat.util.mc
import net.minecraft.client.gui.AbstractGui
import net.minecraft.client.gui.IGuiEventListener
import net.minecraft.client.gui.IRenderable

class ChatHandle(private val chat: ChatBox) : AbstractGui(), IGuiEventListener, IRenderable, ILocatable {

   override val xPos get() = chat.xPos + chat.width - width - 1
   override val yPos get() = chat.yPos + 1
   override val width get() = chat.tray.height - 2
   override val height get() = chat.tray.height - 2

    override fun render(x: Int, y: Int, parTicks: Float) {
        val color = getColor(this.contains(x, y))
        hLine(xPos + 3, xPos + width - 4, yPos + 4, color)
        vLine(xPos + width - 4, yPos + 4, yPos + height - 2, color)
    }

    private fun getColor(hovered: Boolean): Color {
        val opac = (mc.gameSettings.chatOpacity * 255).toInt()
        return argb(opac, 255, 255, if (hovered) 160 else 255)
    }
}
